src/ggrc/assets/javascripts/plugins/ggrc_utils.js

```
@@ -387,6 +387,50 @@
 
       roles.unshift('none');
       return _.max(roles, Array.prototype.indexOf.bind(roleOrder));
+    },
+    /**
+     * A function that returns the Relationship between two objects.
+     *
+     * @param {CMS.Models.Cacheable} first - First object.
+     * @param {CMS.Models.Cacheable} second - Second object.
+     * @param {boolean} rejectOnFail - Flag for reject if not relationship.
+     * @return {Promise} - Resolved with Relationship instance if it was founded
+     * or with undefined if instances not related.
+     */
+    getRelationshipBetweenInstances: function rec(first, second, rejectOnFail) {
+      var relationshipIds = _.intersection(getRelationshipsIds(first),
+        getRelationshipsIds(second));
+      var relationship;
+      var result = $.Deferred();
+
+      function getRelationshipsIds(obj) {
+        var union = _.union(obj.related_sources, obj.related_destinations);
+        return _.map(union, 'id');
+      }
+      if (!relationshipIds.length && rejectOnFail) {
+        return result.resolve();
+      }
+      if (!relationshipIds.length) {
+        // try to refresh the instances if don't find the relationship between them
+        result = $.when(first.refresh(), second.refresh())
+          .then(function (fObj, sObj) {
+            return rec(fObj, sObj, true);
+          }, function (e) {
+            result.reject(e);
+          });
+      } else if (relationshipIds.length === 1) {
+        relationship = CMS.Models.Relationship
+          .findInCacheById(relationshipIds[0]);
+        result.resolve(relationship);
+      } else if (relationshipIds.length > 1) {
+        // Can't be more than one Relationship, therefore this option is unlikely,
+        // and impossible with the correct data. It's just an additional check
+        // and because we take the most relevant Relationship
+        relationship = CMS.Models.Relationship
+          .findInCacheById(_.max(relationshipIds));
+        result.resolve(relationship);
```

@AlexNo It isn't indeed, yes, fixing that would be out of scope. The way I understood @Smotko, though,  was that we should raise an error instead of a mere warning that would allow the user to continue using the application, because the latter could cause further inconsistencies in the DB.
@Smotko Please correct me if I'm wrong.